src/main/java/uk/gov/hmcts/reform/translate/security/idam/IdamRepository.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.reform.translate.security.idam;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+@Component
+@Slf4j
+public class IdamRepository {
+
+    private final IdamClient idamClient;
+
+    @Autowired
+    public IdamRepository(IdamClient idamClient) {
+        this.idamClient = idamClient;
+    }
+
+    @Cacheable("userInfoCache")
```

Comment:
Don't forget to configure caching for using `@Cacheable` to avoid the defaults.

ACA has a property-only config which is simple - [here](https://github.com/hmcts/aac-manage-case-assignment/blob/master/src/main/resources/application.yaml#L26-L29), using [this](https://github.com/hmcts/aac-manage-case-assignment/blob/master/build.gradle#L308) and [this](https://github.com/hmcts/aac-manage-case-assignment/blob/master/build.gradle#L342).

Reply:
👍

Reply:
Fixed in https://github.com/hmcts/ts-translation-service/pull/10/commits/f1be3f52ad0a0ad0bb394f97bfe53b7b9e871815